Sources/TensorFlow/Layers/Normalization.swift

```diff
@@ -12,6 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// Normalizes a tensor by `mean` and `variance`, and applies a `scale` to it, as well as an `offset`.
+@differentiable
+func _batchNorm<Scalar: TensorFlowFloatingPoint>(
```

Comment:
I think this function should be private (and without `_`). Also, wouldn't `normalize` be a better name? It does not apply batch normalization all the time since you use it for group norm (and indirectly layer norm).

Reply:
`_batchNorm` corresponds to [`tf.nn.batch_normalization`](https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/ops/nn_impl.py#L1373-L1434).
I took name from it but `normalize` sounds better.

Reply:
Sounding better is not usually the criterion we're looking for when choosing a name; the first priority is semantic faithfulness.  Is this part of the calculation actually returning a normalized version of its first argument for some common understanding of the meaning of “normalization?”  If so, `normalized` is great.

Reply:
It depends on the shape of parameter tensors If this is `batchNorm` or not, so `normalize` is better.